File: 二分查找/1.小于等于某个数的第一个位置/java/Main.java
import java.io.*;
// 1:无需package
// 2: 类名必须Main, 不可修改

public class Main {

    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length-1;
        int mid;
        while (left<=right){
            mid = left+(right-left)/2;
            if(nums[mid]==target){
                return mid;
            }else if(nums[mid]>target){
                right=mid-1;
            }else{
                left = mid+1;
            }
        }
        return left;
    }

    public static void main(String[] args) throws IOException {
        Main main = new Main();
    }
}

File: 二分查找/1.小于等于某个数的第一个位置/python/main.py
from functools import cmp_to_key
from typing import List


# import sys
#
# sys.setrecursionlimit(100000)  # 例如这里设置为十万


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return left


if __name__ == '__main__':
    s = Solution()
